session/create_producer.go

```diff
@@ -32,25 +33,31 @@ func (producer *createProducer) GetRequestEndpoint() communication.RequestEndpoi
 }
 
 func (producer *createProducer) NewResponse() (responsePtr interface{}) {
```

Comment:
We prob don't need it at all. All structure members are public.. and no need to test then too.

Reply:
Did not get this comment.
But we will need to simplify and refactor communication endpoints somehow

Reply:
We could return structure directly, no need for factory.